src/model.ts

```diff
@@ -694,7 +694,15 @@ export class SpecQueryModel {
 
         // otherwise, assign the proper to the field def
         if (encQ[prop] !== undefined) {
-          fieldDef[prop] = encQ[prop];
+
+          // axis, legend, scale, and sort only support certain channels
+          let propNeedsToCheckChannel: boolean = contains([Property.AXIS, Property.LEGEND, Property.SCALE, Property.SORT], prop);
+
+          // We only output axis, legend, scale, or sort that have correct supporting channel
+          if ((propNeedsToCheckChannel && !isEnumSpec(encQ.channel) && PROPERTY_SUPPORTED_CHANNELS[prop][encQ.channel as Channel]) ||
```

Comment:
Looks like you're excluding all these property is channel is an enumSpec.  
Add a test case for that and fix this!

Use 
```
if (isEnumSpec(encQ.channel) ||  // channel is enum spec, so the property might be supported 
  !PROPERTY_SUPPORTED_CHANNELS[prop] ||  // all channel support this prop
  PROPERTY_SUPPORTED_CHANNELS[prop][encQ.channel as Channel])
) {
  ...
}
```

and you can get rid of `propNeedsToCheckChannel` call that uses `contains`, which is unnecessary O(n).